examples/pyomobook/test_book_examples.py

```diff
@@ -0,0 +1,384 @@
+            for i in solvers_:
+                if not solver_available[i]:
+                    _missing.append(i)
+            return "Solver%s %s %s not available" % (
+                's' if len(_missing) > 1 else '',
+                ", ".join(_missing),
+                'are' if len(_missing) > 1 else 'is',)
+
+    if name in package_dependencies:
+        packages_ = package_dependencies[name]
+        if not all([package_available[i] for i in packages_]):
+            # Skip the test because a package is not available
+            _missing = []
+            for i in packages_:
+                if not package_available[i]:
+                    _missing.append(i)
+            return "Package%s %s %s not available" % (
+                's' if len(_missing) > 1 else '',
+                ", ".join(_missing),
+                'are' if len(_missing) > 1 else 'is',)
+
+        # This is a hack, xlrd dropped support for .xlsx files in 2.0.1 which
+        # causes problems with older versions of Pandas<=1.1.5 so skipping
+        # tests requiring both these packages when incompatible versions are found
+        if 'pandas' in package_dependencies[name] and 'xlrd' in package_dependencies[name]:
+            if check_min_version(package_modules['xlrd'], '2.0.1') and \
+               not check_min_version(package_modules['pandas'], '1.1.6'):
+                return "Incompatible versions of xlrd and pandas"
+
+    return False
+
+def filter(line):
+    """
+    Ignore certain text when comparing output with baseline
+    """
+    for field in ( '[',
+                   'password:',
+                   'http:',
+                   'Job ',
+                   'Importing module',
+                   'Function',
+                   'File', 
+                   'Restricted license',
+                   'Using license',
+                   'Matplotlib',
+                   '    ^'):
+        if line.startswith(field):
+            return True
+    for field in ( 'Total CPU',
+                   'Ipopt',
+                   'Status: optimal',
+                   'Status: feasible',
+                   'time:',
+                   'Time:',
+                   'with format cpxlp',
+                   'usermodel = <module',
+                   'execution time=',
+                   'Solver results file:',
+                   'TokenServer'):
+        if field in line:
+            return True
+    return False
+
+
+def filter_file_contents(lines):
+    filtered = []
+    for line in lines:
+        if not line or filter(line):
+            continue
+
+        items = line.strip().split()
+        for i in items:
+            if not i:
+                continue
+            if not (i.startswith('/') or i.startswith(":\\", 1)):
+                try:
+                    filtered.append(float(i))
+                except:
+                    filtered.append(i)
+
+    return filtered
+
+
+py_test_tuples=[]
+sh_test_tuples=[]
+
+for tdir in testdirs:
```

Comment:
`testdirs` is defined as `[currdir,]`. Is there a possibility that `testdirs` is something other than a single-item list? If not, I feel like the outer loop is unnecessary.

Reply:
Good point, I'll make that change.